### DISCUSSION TITLE
Get report automatically

### DISCUSSION THREAD

**Discussion by GilFernandes2000** [DefectDojo/django-DefectDojo#9898] [Q&A] [answered]
Hi,

I'm doing some automation with DefectDojo.
The last step of the automation is getting a report from DefectDojo and sending it through email.
The only solution I've got is getting the json through the API and then building the report from there.
Is there a way to get and download a report from a product automatically without having to build the report from the json?

Thank you all in advance.

**Comment by mtesauro** [comment, marked as answer]
FWIW, I've always done the API to report route since you have complete control of what and how the email report looks.

I guess you could using something like Playwright to automate the UI to create, download and send a report via email but that feels like it would be more work then just calling the API.

You might also look to see if any of the existing alerts/notifications would work for your use case as those can be emailed already after you setup SMTP for your install.

**Reply by GilFernandes2000** [reply]
Hi,

Thanks for the clarification.

**Reply by topriko** [reply]
@GilFernandes2000 Hi I'm also trying to build something similar how far have you gotten maybe we can work together on report automation

**Reply by GilFernandes2000** [reply]
Hi @Rikozi, sorry for the late reply. I followed @mtesauro advice and got the information from the API and then trimmed it to the point where I got what I wanted. I don't know what the advancements are in this topic but I think a good idea could be to add filters to the API call to get what you wanted directly and not having to do it manually. Other than that we could create a PDF report template and suggest it to the DefectDojo team.
